WordPress/openverse-catalog
# Improved data refresh status reporting

## Fixes


Fixes #699 by @stacimc

## Description


This PR adds additional slack notifications for the first few steps of the data refresh, namely the matview/popularity table refreshes. These messages are set up to mirror the exact format of the ingestion server's messages, so they appear seamless.

I ended up wrapping the `_month_check` function to make reporting easier, since it's possible that it exits early and I didn't want to add a slack message step before every `return`. This was the most straightforward solution for the DAG since that is a branching operator. For all other instances, I added another `PythonOperator` step which issued a reporting slack message.

**DAG**
![Screenshot_2022-09-27_10-53-30](https://user-images.githubusercontent.com/10214785/192602007-7ffb2bca-8b3c-4b9a-b766-7328321ecd82.png)


**Example messages**
![Screenshot_2022-09-27_10-54-21](https://user-images.githubusercontent.com/10214785/192602026-2167894f-e831-45f6-b1a8-2b8d298f2b9d.png)


## Testing Instructions

As always, `just test`. If you want to test the reporting too, add `AIRFLOW_VAR_SLACK_MESSAGE_OVERRIDE=true` to your `.env` file then run the following (note that you will need to spin up the API stack locally in order to test the full data refresh).

1. Start a data refresh. Since this is the first run in the month, you should see a total of 4 messages.
1. Once the previous data refresh is done, start a second data refresh run. This should only send 2 messages.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
